src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentNamespaceDirective.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components
+{
+    internal class ComponentNamespaceDirective
+    {
+        public static readonly DirectiveDescriptor Directive = DirectiveDescriptor.CreateDirective(
+            "namespace",
+            DirectiveKind.SingleLine,
+            builder =>
+            {
+                builder.AddNamespaceToken(
+                    ComponentResources.NamespaceDirective_NamespaceToken_Name,
+                    ComponentResources.NamespaceDirective_NamespaceToken_Description);
+                builder.Usage = DirectiveUsage.FileScopedSinglyOccurring;
+                builder.Description = ComponentResources.NamespaceDirective_Description;
+            });
+
+        public static RazorProjectEngineBuilder Register(RazorProjectEngineBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.AddDirective(Directive, FileKinds.Component, FileKinds.ComponentImport);
+            return builder;
```

Comment:
Can't follow the usual pattern we have for directive passes here because `DirectiveClassifierPass` runs much later in the pipeline but I need to visit the namespace directives in the tag helper binding phase.

Reply:
How bad would it be if we just changed the other namespace directive to no longer act as a `DirectiveClassifierPass` and follow the same pattern as components? Asking because it'd be awesome if we could just have 1 mechanism for setting namespaces instead of the fork in when they're set amongst the two file kinds.

Reply:
I agree. Things like namespace are fundamental.

Reply:
Okay I'll try to combine both of those.